gui/src/components/ProviderTheme.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  PaletteMode,
+  Theme,
+  ThemeProvider,
+  createTheme,
+  useMediaQuery,
+} from "@mui/material";
+import { grey } from "@mui/material/colors";
+import { invoke } from "@tauri-apps/api/tauri";
+import { useRegisterActions } from "kbar";
+import { ReactNode, createContext, useMemo } from "react";
+
+import { useInvoke } from "../hooks/tauri";
+import { GeneralSettings } from "../types";
+
+const themeModeId = "themeMode";
+
+const getDesignTokens = (mode: PaletteMode) => ({
+  palette: {
+    mode,
+    ...(mode === "light"
+      ? {
+          background: {
+            default: grey[50],
+          },
+        }
+      : {
+          background: {},
+        }),
+  },
+});
+
+export const ThemeContext = createContext<Theme | null>(null);
+
+export function ProviderTheme({ children }: { children: ReactNode }) {
```

Comment:
maybe I'm confused but shouldn't the name be `ThemeProvider` instead?

Reply:
I like to use prefixes instead so all providers live side-by-side in the file system. Do you think that's weird?

Reply:
it's fine with me, now that I know the reason. I honestly thought it was by mistake 😅 

but since you mentioned it: wouldn't a more appropriate solution be to have a `Providers` directory instead of a flat `components`?

(not requesting changes here, consider this one solved)